theLudovyc/Resources/Resources.gd

```diff
@@ -1,18 +1,36 @@
 extends Object
 class_name Resources
 
-enum Types { Wood, Textile }
+enum Types { Wood, Textile, Board }
+enum LevelTypes { Gathered, TransformedOnce, TransformedTwice }
+enum StorageType { Building_Input, Building_Output }
+enum Data {StorageType, LevelType, Icon}
 
-const Icons = {
-	Types.Wood: preload("res://Art/Image/Gui/Icons/Resources/32/008.png"),
-	Types.Textile: preload("res://Art/Image/Gui/Icons/Resources/32/003.png")
+const AllResources: Dictionary = {
+	Types.Wood:    {
+		Data.StorageType: StorageType.Building_Input,
+		Data.LevelType: LevelTypes.Gathered,
+		Data.Icon: preload("res://Art/Image/Gui/Icons/Resources/32/008.png")
+	},
+	Types.Textile: {
+		Data.StorageType: StorageType.Building_Output,
+		Data.LevelType:LevelTypes.TransformedTwice,
+		Data.Icon: preload("res://Art/Image/Gui/Icons/Resources/32/003.png")
+	},
+	Types.Board: {
+		Data.StorageType: StorageType.Building_Output,
+		Data.LevelType:LevelTypes.TransformedOnce,
+		Data.Icon: preload("res://Art/Image/Gui/Icons/Resources/32/004.png")
+	},
 }
 
+static func has_rescource(resource_type: Types) -> bool:
+	return AllResources.has(resource_type)
 
 static func get_resource_icon(resource_type: Types) -> Texture2D:
-	return Icons.get(resource_type)
-
-
-enum LevelTypes { Gathered, TransformedOnce, TransformedTwice }
+	return AllResources.get(resource_type)[2]
```

Comment:
Regroup the datas is a good idea. But here, you do not handle the case where resource_type can be not found

Reply:
To be fair - your code did not either ;)

What should be the outcome if the type is not found? A dummy icon?

Reply:
I do this with
```
static func get_resource_icon(resource_type: Types) -> Texture2D:
  return Icons.get(resource_type)
```
From the doc about Dictionary.get(...) `If the key does not exist, returns default, or null if the parameter is omitted.`

In your proposal you can call [2] on a null value